### ISSUE TITLE
FR: require_admin and /or other user view settings

### ISSUE THREAD

**Issue by Mariusthvdb** [music-assistant/hass-music-assistant#245]
currently, all logged users in my HA instance see the Music assistant side panel.

would love to be able to select which users can see that.. Not having my music being played by others feels like a fundamental security setting?

thanks for considering

**Comment by marcelveldt** [comment]
I do not consider the ability of playing music a security setting but a feature but that's just personal I guess ;-)
The ability to show/hide panels is not something each integration should implement (not even custom integrations) but imo this should be implemented at core level, for example with user roles to which you assign certain functionality.

So, basically my answer is that I will not consider adding this feature request to Music Assistant but I'm willing to start the architecture discussion for core ;-)

**Comment by Mariusthvdb** [comment]
clearly you dont have kids old enough to be little teasers... ;-)

thanks if you open the discussion though, that would be really helpful!

**Comment by marcelveldt** [comment]
haha, oh yes I do, and I find it kind of fun to teach them how to handle IT instead of hiding it from them ;-)
Although I do not agree with an approach of hiding stuff because someone in my household may abuse it, I do agree that it is a good UX to hide stuff that is not relevant for some persons in the house. That's why I think that role bases access stuff is still a good idea but no so much for security (I want my HA to be secured) but for usability reasons.

For example I'm interested in nitty pitty details about power consumption and automations performance and my wife just needs a few big buttons to turn off a light.

I'll let you know once I've prepared something on the architecture repo.

**Comment by Mariusthvdb** [comment]
thanks!

btw, I wasnt perse talking 'little' kids.....And I thought them all to well ;-)

 when they're old enough to travel the world and party, I've learned that remote access to my home isn't perse a good thing... even though I understand the fun, I'd like to be in charge who controls my music playing in the middle of the night over the broadcast system.

For now I've used the generic setting on users requiring local access, but thats very drastic ofc. more fine grained control would be appreciated. 

Feel free to close here.
